staker/stateless_block_validator.go

```diff
@@ -305,6 +308,13 @@ func (v *StatelessBlockValidator) ValidationEntryRecord(ctx context.Context, e *
 				return err
 			}
 		}
+		if usingEspresso {
```

Comment:
4. Finally, we can add the preimage here by calling the inbox oracle method we added. If the HotShot header isn't available, `e.State` will not be set to ready below. This may be how validation is delayed. Not sure about the specific retry logic yet.